pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -798,7 +800,7 @@ public void unloadNamespaceBundlesGracefully() {
                 });
             } else {
                 return topics.computeIfAbsent(topic, (topicName) -> {
-                    if (createIfMissing) {
+                    if (createIfMissing || TopicName.get(topicName).isPartitioned()) {
```

Comment:
I think it's not a correct fix. The expected behavior is if the partitioned non-persistent topic created, then we allow the partitions of the partitioned non-persistent topic to create automatically

Reply:
You're right @codelipenghui I'm fixing it (y). Thanks!

Reply:
A new version checked in